app/database/server/models/team.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * A Team houses and enables communication to happen across channels and users.
+ */
+export default class Team extends Model {
+    /** table (entity name) : Team */
+    static table = TEAM;
+
+    /** associations : Describes every relationship to this entity. */
+    static associations: Associations = {
+
+        /** A TEAM has a 1:N relationship with CHANNEL. A TEAM can possess multiple channels */
+        [CHANNEL]: {type: 'has_many', foreignKey: 'team_id'},
+
+        /** A TEAM has a 1:N relationship with GROUPS_IN_TEAM. A TEAM can possess multiple groups */
+        [GROUPS_IN_TEAM]: {type: 'has_many', foreignKey: 'team_id'},
+
+        /** A TEAM has a 1:1 relationship with MY_TEAM. */
+        [MY_TEAM]: {type: 'has_many', foreignKey: 'team_id'},
+
+        /** A TEAM has a 1:N relationship with SLASH_COMMAND. A TEAM can possess multiple slash commands */
+        [SLASH_COMMAND]: {type: 'has_many', foreignKey: 'team_id'},
+
+        /** A TEAM has a 1:N relationship with TEAM_CHANNEL_HISTORY. A TEAM can possess multiple channels histories*/
+        [TEAM_CHANNEL_HISTORY]: {type: 'has_many', foreignKey: 'team_id'},
+
+        /** A TEAM has a 1:N relationship with TEAM_MEMBERSHIP. A TEAM can regroup multiple users */
+        [TEAM_MEMBERSHIP]: {type: 'has_many', foreignKey: 'team_id'},
+
+        /** A TEAM has a 1:N relationship with TEAM_SEARCH_HISTORY. A TEAM can possess multiple channels recently visited*/
+        [TEAM_SEARCH_HISTORY]: {type: 'has_many', foreignKey: 'team_id'},
+    };
+
+    /** allow_open_invite : Boolean flag indicating if this team is open to the public */
+    @field('allow_open_invite') allowOpenInvite!: boolean;
+
+    /** description : The description for the team */
+    @field('description') description!: string;
+
+    /** display_name : The display name for the team */
+    @field('display_name') displayName!: string;
+
+    /** is_group_constrained : Boolean flag indicating if members are managed groups */
+    @field('is_group_constrained') isGroupConstrained!: boolean;
+
+    /** last_team_icon_updated_at : Timestamp for when this team's icon has been updated last */
+    @field('last_team_icon_updated_at') lastTeamIconUpdatedAt!: number;
+
+    /** name : The name for the team */
+    @field('name') name!: string;
+
+    /** type : The type of team ( e.g. open/private ) */
+    @field('type') type!: string;
+
+    /** allowed_domains : List of domains that can join this team */
+    @json('allowed_domains', (rawJson) => rawJson) allowedDomains!: string;
+
+    /** channels : All the channels associated with this team */
+    @children(CHANNEL) channels!: Channel[];
+
+    /** groupsInTeam : All the groups associated with this team */
+    @children(GROUPS_IN_TEAM) groupsInTeam!: GroupsInTeam[];
+
+    /** myTeam : Lazy query property returning only the team that this user is part of  */
+    @lazy myTeam = this.collections.get(MY_TEAM).query(Q.on(TEAM, 'id', this.id)) as Query<MyTeam>;
```

Comment:
The `Team` and `MyTeam` entities share a 1:1 relationship with each other.  `MyTeam` is an extension of the `Team` entity except it will hold only the information to the teams that a user belongs to.

In WDB, we found it a bit difficult to describe the 1:1 relationship.  On one end you would have 'has_many' association (e.g. for Team) and on the other end you would have a 'belongs_to' association (e.g. MyTeam).  

So we decided to use a lazy query in the 'parent' entity and a relation on the child side.   If we didn't use the lazy query, we would have got a `@children` field which is  a bit counter-intuitive for a single child relation.

Reply:
Sure thing.   Queries in WDB are lazily loaded; it's only when we call them from within a component that `user.myTeam` will get a value.

So in short, we'll either connect a component using the HOC `withObservables` or import the `database` reference and call a specific `collections` and then do a fetch operation on a query.  You actually came across the `flatlist` component in my other PR and in there it was using a list of servers stored in the db.

Queries can either be observed in the `withObservables` HOC or directly fetched.  See : 
https://nozbe.github.io/WatermelonDB/Query.html#executing-queries